components/brave_wallet_ui/common/constants/persisted-state-keys-whitelists.ts

```diff
@@ -0,0 +1,233 @@
+  Exclude<keyof ApiSliceRootState, BlacklistedApiSliceRootStateKeys>
+> = [
+  'config', // persist the cache configuration
+  'queries' // fulfilled (whitelisted) queries
+]
+
+type BlacklistedWalletApiQueryEndpointName = Subset<
+  WalletApiQueryEndpointName,
+  | 'getAllPendingTransactions' // prevent tampering with Pending Txs
+  | 'getDefaultAccountAddresses' // defaults may have changed since last launch
+  | 'getDefaultFiatCurrency' // defaults may have changed since last launch
+  | 'getGasEstimation1559' // network fees constantly change
+  | 'getSelectedAccountAddress' // selections may have changed
+  | 'getSelectedChain' // selections may have changed
+  | 'getSelectedPendingTransactionId' // selections may have changed
+  | 'getSolanaEstimatedFee' // network fees constantly change
+>
+
+type WhitelistedWalletApiQueryEndpointName = Exclude<
+  WalletApiQueryEndpointName,
+  BlacklistedWalletApiQueryEndpointName
+>
+
+const apiEndpointCacheWhitelist: WhitelistedWalletApiQueryEndpointName[] = [
+  'getAccountInfosRegistry', // persist wallet addresses
+  'getAccountTokenCurrentBalance', // persist current balances
+  'getAddressByteCode', // persist contract bytecode lookups
+  'getAllTransactionInfosForAddressCoinType', // persist completed txs for address
+  'getAllTransactionsForAddressCoinType', // persist completed txs for address
+  'getCombinedTokenBalanceForAllAccounts', // persist current combined balances
+  'getERC721Metadata', // persist nft metadata
+  'getNetworksRegistry', // persist list of networks
+  'getSwapSupportedNetworkIds', // persist which networks support Brave Swap
+  'getTokenBalancesForChainId', // persist token balances on each chain
+  'getTokenSpotPrice', // persist token spot price
+  'getTokensRegistry', // persist known tokens registry
+  'getUserTokensRegistry', // persist user tokens registry
+]
+
+type BlacklistedPageStateKey = Subset<
+  keyof PageState,
+  | 'enablingAutoPin' // unused state
+  | 'hasInitialized' // unused state
+  | 'importAccountError' // don't persist import errors
+  | 'importWalletAttempts' // do not rely on persisted storage for attempts
+  | 'importWalletError' // don't persist import errors
+  | 'invalidMnemonic' // don't persist errors
+  | 'isAutoPinEnabled' // selection may have changed
+  | 'isCryptoWalletsInitialized' // selection may have changed
+  | 'isImportWalletsCheckComplete' // importable wallets may have changed
+  | 'isLocalIpfsNodeRunning' // selection may have changed
+  | 'isMetaMaskInitialized' // selection may have changed
+  | 'mnemonic' // do not store private data
+  | 'nftMetadataError' // do not persist errors
+  | 'nftsPinningStatus' // pinning may have been disabled since last launch
+  | 'pinStatusOverview' // unused state
+  | 'setupStillInProgress' // start onboarding again if not completed
+  | 'showRecoveryPhrase' // do not show private data on app relaunch
+>
+
+export const pageStatePersistorWhitelist: Array<
+  Exclude<
```

Comment:
This approach appears to be creating a blocklist here still. Can we make this an allowlist again?

The reason I'm thinking this is the case is because the pageStatePersistorWhitelist is generated by first taking the `PageState` and then excluding the `BlacklistedPageStateKey` from the PageState. Instead, I believe we should be defining this via using `Subset<keyof PageState, {UnionType of explicitly allowed PageStates}>`

Reply:
Oh wait, are we doing it this way so that the typing system knows to explicitly never accidently include one of the items on the `BlacklistedPageStateKey`? What happens if the `PageState` type contains a value that's not on either `pageStatePersistorWhitelist` or `BlacklistedPageStateKey`?

Reply:
correct, the type system is enforcing that only keys from the state object that have not been blacklisted can be added to the whitelist without throwing a type error. This was to provide a way to provide explicit reasoning as to why properties were whitelisted or blacklisted. The blacklists are types only, and will not be used during runtime

Reply:
Cool that makes sense and seems like the best way to leverage the type system. Thanks for clarifying that. I see no new concerns with sec review side of things.